File: src/main/java/com/api/twitter/security/application/usecases/DeleteUserCredentialsUseCase.java
package com.api.twitter.security.application.usecases;

import com.api.twitter.security.infrastructure.persistence.UserCredentialsRepository;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class DeleteUserCredentialsUseCase {
    private UserCredentialsRepository userCredentialsRepository;

    public DeleteUserCredentialsUseCase(UserCredentialsRepository userCredentialsRepository) {
        this.userCredentialsRepository = userCredentialsRepository;
    }

    public void execute(UUID userId) {
        userCredentialsRepository.deleteById(userId);
    }
}

File: src/main/java/com/api/twitter/security/application/usecases/DeleteUserCredentialsUseCase.kt
package com.api.twitter.security.application.usecases

import com.api.twitter.security.infrastructure.persistence.UserCredentialsRepository
import org.springframework.stereotype.Component
import java.util.*

@Component
class DeleteUserCredentialsUseCase(private val userCredentialsRepository: UserCredentialsRepository) {
    fun execute(userId: UUID) {
        userCredentialsRepository.deleteById(userId)
    }
}
